src/solvers/dgsem_p4est/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,433 @@
+
+        gradient_x_node = Ja11 * gradients_reference_1 + Ja21 * gradients_reference_2
+        gradient_y_node = Ja12 * gradients_reference_1 + Ja22 * gradients_reference_2
+
+        set_node_vars!(gradients_x, gradient_x_node, equations_parabolic, dg, i, j, element)
+        set_node_vars!(gradients_y, gradient_y_node, equations_parabolic, dg, i, j, element)
+      end
+
+    end
+  end
+
+  # Prolong solution to interfaces
+  @trixi_timeit timer() "prolong2interfaces" prolong2interfaces!(
+    cache_parabolic, u_transformed, mesh, equations_parabolic, dg.surface_integral, dg)
+
+  # Calculate interface fluxes
+  @trixi_timeit timer() "interface flux" begin
+    (; surface_flux_values) = cache_parabolic.elements
+    (; neighbor_ids, node_indices) = cache.interfaces
+    index_range = eachnode(dg)
+    index_end = last(index_range)
+
+    @threaded for interface in eachinterface(dg, cache)
+      # Get element and side index information on the primary element
+      primary_element = neighbor_ids[1, interface]
+      primary_indices = node_indices[1, interface]
+      primary_direction = indices2direction(primary_indices)
+
+      # Create the local i,j indexing on the primary element used to pull normal direction information
+      i_primary_start, i_primary_step = index_to_start_step_2d(primary_indices[1], index_range)
+      j_primary_start, j_primary_step = index_to_start_step_2d(primary_indices[2], index_range)
+
+      i_primary = i_primary_start
+      j_primary = j_primary_start
+
+      # Get element and side index information on the secondary element
+      secondary_element = neighbor_ids[2, interface]
+      secondary_indices = node_indices[2, interface]
+      secondary_direction = indices2direction(secondary_indices)
+
+      # Initiate the secondary index to be used in the surface for loop.
+      # This index on the primary side will always run forward but
+      # the secondary index might need to run backwards for flipped sides.
+      if :i_backward in secondary_indices
+        node_secondary = index_end
+        node_secondary_step = -1
+      else
+        node_secondary = 1
+        node_secondary_step = 1
+      end
+
+      for node in eachnode(dg)
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u,
+                                           equations_parabolic, dg, node, interface)                                              
+        flux = 0.5 * (u_ll + u_rr)
+
+        for v in eachvariable(equations_parabolic)
+          surface_flux_values[v, node, primary_direction, primary_element] = flux[v]
+          surface_flux_values[v, node_secondary, secondary_direction, secondary_element] = flux[v]
+        end
+    
+        # Increment primary element indices to pull the normal direction
+        i_primary += i_primary_step
+        j_primary += j_primary_step
+        # Increment the surface node index along the secondary element
+        node_secondary += node_secondary_step
```

Comment:
Isn't all this identical to the non-parabolic code? Can we reuse that?

Reply:
Good question. There's a very slight difference in https://github.com/jlchan/Trixi.jl/blob/e072790342a9b49980bbe423ee7766095ed0a713/src/solvers/dgsem_p4est/dg_2d.jl#L189-L192
```julia
  for v in eachvariable(equations)
    surface_flux_values[v, primary_node_index, primary_direction_index, primary_element_index] = flux_[v]
    surface_flux_values[v, secondary_node_index, secondary_direction_index, secondary_element_index] = -flux_[v]
  end
```
There's no negative on the second flux term in this implementation. 

However, we can reuse most of it and just specialize on `calc_interface_flux!` in https://github.com/jlchan/Trixi.jl/blob/e072790342a9b49980bbe423ee7766095ed0a713/src/solvers/dgsem_p4est/dg_2d.jl#L175-L193 instead. I'll try that.